### PR TITLE
Retain pure annotations in class fields

### LINKED ISSUE #3591: error when in __PURE__ in class static field
- Rollup Version: 2.10.7
- Operating System (or Browser): Windows 10
- Node Version (if applicable): 12
- Link to reproduction (IMPORTANT, read below):

https://rollupjs.org/repl/?version=2.10.9&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGRlZmF1bHQlMjBjbGFzcyUyMCU3QiU1Q24lMjAlMjAlMjAlMjBzdGF0aWMlMjBrJTIwJTNEJTIwJTJGKiUyM19fUFVSRV9fKiUyRlYoKSUzQiU1Q24lN0QlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

main.js:

```js
export default class {
    static k = /*#__PURE__*/V();
};
```

### Expected Behavior

no error

### Actual Behavior

```
TypeError: base$1[type] is not a function
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.base$1.ClassBody                                              (/rollup/dist/shared/rollup.js:14831:5)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.base$1.Class                                                  (/rollup/dist/shared/rollup.js:14824:3)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.base$1.ClassDeclaration.base$1.ClassExpression                (/rollup/dist/shared/rollup.js:14820:84)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.skipThrough                                                   (/rollup/dist/shared/rollup.js:14578:37)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.base$1.ExportNamedDeclaration.base$1.ExportDefaultDeclaration (/rollup/dist/shared/rollup.js:14796:7)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.skipThrough                                                   (/rollup/dist/shared/rollup.js:14578:37)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at Object.base$1.Program.base$1.BlockStatement                          (/rollup/dist/shared/rollup.js:14590:5)
    at handlePureAnnotationsOfNode                                          (/rollup/dist/shared/rollup.js:14847:21)
    at markPureCallExpressions                                              (/rollup/dist/shared/rollup.js:14867:5)
    at Module.setSource                                                     (/rollup/dist/shared/rollup.js:15416:13)
    at ModuleLoader.addModuleSource                                         (/rollup/dist/shared/rollup.js:17457:20)
    at async ModuleLoader.fetchModule                                       (/rollup/dist/shared/rollup.js:17518:9)
    at async Promise.all                                                    (index 0)
    at async Promise.all                                                    (index 0)
    at async getCombinedPromise                                             (/rollup/dist/shared/rollup.js:17477:13)
    at async ModuleLoader.awaitLoadModulesPromise                           (/rollup/dist/shared/rollup.js:17482:9)
    at async ModuleLoader.addEntryModules                                   (/rollup/dist/shared/rollup.js:17378:33)
    at async Promise.all                                                    (index 0)
    at async Graph.parseModules                                             (/rollup/dist/shared/rollup.js:18589:63)
    at async Graph.build                                                    (/rollup/dist/shared/rollup.js:18449:61)
    at async rollupInternal                                                 (/rollup/dist/shared/rollup.js:18840:18)
```

### PR DESCRIPTION
**Pull request by lukastaegert**
This PR contains:
- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [x] yes (*bugfixes and features will not be merged without tests*)
- [ ] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers:
Resolves #3591 

### Description
Unfortunately, `acorn-walk` is not yet properly updated for the new nodes added for class fields. This adds a temporary patch until this is hopefully resolved once the feature is merged to acorn.

### CODE REVIEW

File: src/utils/pureComments.ts [hunk 1]
```diff
@@ -3,6 +3,14 @@ import * as acorn from 'acorn';
 import { base as basicWalker } from 'acorn-walk';
 import { CommentDescription } from '../Module';
 
+// patch up acorn-walk until class-fields are officially supported
+basicWalker.FieldDefinition = function (node: any, st: any, c: any) {
+	c(node.key, st, 'Expression');
```
Comment (LongTengDao) [inline src/utils/pureComments.ts:8]: Shouldn't this line to be `if ( node.computed ) { c(node.key, st, 'Expression'); }`, just like `MethodDefinition` / `Property` do in `acorn-work`?

File: test/form/samples/pure-class-field/_expected.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class main {
+	static k = /*#__PURE__*/ V();
+	static /*#__PURE__*/ ignored;
```
Comment (LongTengDao) [inline test/form/samples/pure-class-field/_expected.js:3]: Does this sample line has special meaning in rollup? It seems not a pure mark for function call.
Is it the new feature of rollup?
Or just as a test, to make sure here is no bug, without any other use?

File: src/utils/pureComments.ts (hunk 1 above)
Comment (lukastaegert) [reply src/utils/pureComments.ts:8]: Good catch, was obviously copying without thinking.

File: test/form/samples/pure-class-field/_expected.js (hunk 2 above)
Comment (lukastaegert) [reply test/form/samples/pure-class-field/_expected.js:3]: The latter, a test to cover a branch in the logic even if it makes no sense to do so in real code.

**Pull request merged**